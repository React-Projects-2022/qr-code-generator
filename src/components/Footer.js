export const Footer = () => (
  <div class="footer">
    <p>QR Code Generator - Anartz Mugika Ledo - { new Date().getFullYear()}</p>
  </div>
);
