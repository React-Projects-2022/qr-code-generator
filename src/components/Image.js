export const Image = ({ qrCode, text }) => (
  <img src={qrCode} alt={"QR Code Image with ".concat(text)} />
);
