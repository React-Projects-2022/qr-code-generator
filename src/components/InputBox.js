export const InputBox = ({
  size,
  temp,
  changeSize,
  changeBgColor,
  asignInputTempText,
  asignFindText,
}) => {
  const handleClick = () => asignFindText(temp);
  return (
    <div className="input-box">
      <div className="gen">
        <input
          type="text"
          autoFocus={true}
          onChange={(e) => {
            asignInputTempText(e.target.value);
          }}
          placeholder="Enter text to encode"
        />
        <button className="button" onClick={handleClick}>
          Generate
        </button>
      </div>
      <div className="extra">
        <h5>Background Color:</h5>
        <input
          type="color"
          onChange={(e) => {
            changeBgColor(e.target.value.substring(1));
          }}
        />
        <h5>Dimension:</h5>
        <input
          type="range"
          min="200"
          max="600"
          value={size}
          onChange={(e) => {
            changeSize(e.target.value);
          }}
        />
      </div>
    </div>
  );
};
