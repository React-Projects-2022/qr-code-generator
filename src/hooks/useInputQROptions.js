import { useState } from "react";

export const useInputQROptions = () => {
  const [temp, setTemp] = useState("");
  const [text, setText] = useState("");
  const [size, setSize] = useState(400);
  const [bgColor, setBgColor] = useState("ffffff");

  const changeSize = (sizeValue) => setSize(sizeValue);
  const changeBgColor = (bgColorValue) => setBgColor(bgColorValue);
  const asignInputTempText = (tempTextValue) => setTemp(tempTextValue);
  const asignFindText = (textFinalValue) => setText(textFinalValue);
  return {
    size,
    temp,
    text,
    bgColor,
    changeSize,
    changeBgColor,
    asignInputTempText,
    asignFindText,
  };
};
