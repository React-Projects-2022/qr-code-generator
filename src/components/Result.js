import { Image } from "./Image";
import { downloadImage } from "./../helpers/download-image";
export const Result = ({ qrCode, text }) => (
  <div className="output-box">
    <Image qrCode={qrCode} text={text} />
    <button onClick={() => downloadImage(qrCode, text)} type="button">
      Download
    </button>
  </div>
);
