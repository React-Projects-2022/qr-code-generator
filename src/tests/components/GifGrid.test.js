import { GifGrid } from "../../components/GifGrid";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
// Mockeamos el hooks para añadirle la respuesta que queremos
jest.mock("../../hooks/useFetchGifs");
describe("Pruebas en GifGrid", () => {
  const category = "angular";

  test("De de mostrar que está cargando", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Comprobar si el título de "category" es correcto', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    const title = wrapper.find("h3").text().trim();
    expect(title).toBe(category);
  });

  test("Debe de mostrar items cuando se cargan imágenes con useFetchGifs", () => {
    // Mock con un resultado
    const gifs = [
      {
        id: "sjsjs",
        url: "https://localhost/image/01.jpg",
        title: "Gif angular",
      },
      {
        id: "sjsjs2",
        url: "https://localhost/image/02.jpg",
        title: "Gif angular 2",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    // Si tenemos datos, "p" ya no existe
    const pTag = wrapper.find("p");
    expect(pTag.length).toBe(0);
    // No existe, sería similar al anterior
    expect(pTag.exists()).toBe(false);

    // Evaluando si existe el componente "GridItem"
    expect(wrapper.find("GifGridItem").exists()).toBe(true);
    // Otra manera para evaluar las veces que aparece "GifGridItem"
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
