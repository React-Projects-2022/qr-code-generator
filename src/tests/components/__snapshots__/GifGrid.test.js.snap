// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid De de mostrar que está cargando 1`] = `
<Fragment>
  <h3>
    angular
  </h3>
  <div
    className="card-grid"
  >
    <p>
      Cargando imágenes 
      angular
    </p>
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Debe de mostrar items cuando se cargan imágenes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    angular
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      img={
        Object {
          "id": "sjsjs",
          "title": "Gif angular",
          "url": "https://localhost/image/01.jpg",
        }
      }
      key="sjsjs"
    />
    <GifGridItem
      img={
        Object {
          "id": "sjsjs2",
          "title": "Gif angular 2",
          "url": "https://localhost/image/02.jpg",
        }
      }
      key="sjsjs2"
    />
  </div>
</Fragment>
`;
